Library/Homebrew/utils/curl.rb

```diff
@@ -1,42 +1,46 @@
 require "pathname"
 require "open3"
 
-def curl_args(options = {})
+def curl_executable
   curl = Pathname.new ENV["HOMEBREW_CURL"]
   curl = Pathname.new "/usr/bin/curl" unless curl.exist?
-  raise "#{curl} is not executable" unless curl.exist? && curl.executable?
+  return curl if curl.executable?
+  raise "#{curl} is not executable"
+end
 
+def curl_args(*extra_args, show_output: false, user_agent: :default)
   args = [
-    curl.to_s,
-    "--remote-time",
-    "--location",
+    curl_executable.to_s,
+    "--fail",
   ]
 
-  case options[:user_agent]
-  when :browser
-    args << "--user-agent" << HOMEBREW_USER_AGENT_FAKE_SAFARI
+  args << "--user-agent" << case user_agent
+  when :browser, :fake
+    HOMEBREW_USER_AGENT_FAKE_SAFARI
+  when :default
+    HOMEBREW_USER_AGENT_CURL
   else
-    args << "--user-agent" << HOMEBREW_USER_AGENT_CURL
+    user_agent
   end
 
-  unless options[:show_output]
+  unless show_output
     args << "--progress-bar" unless ARGV.verbose?
     args << "--verbose" if ENV["HOMEBREW_CURL_VERBOSE"]
-    args << "--fail"
-    args << "--silent" if !$stdout.tty? || ENV["TRAVIS"]
   end
 
-  args += options[:extra_args] if options[:extra_args]
-  args
+  args << "--silent" if !$stdout.tty? || ENV["TRAVIS"]
```

Comment:
`--silent` doesn't silence the output, it only silences the progress bar/indicator, which would go to `stderr`, so it doesn't have an effect on `curl_output`.